app/src/main/java/ch/epfl/sweng/studdybuddy/core/User.java

```diff
@@ -29,10 +38,20 @@ public User(String name, ID<User> userId)
         this.favoriteLanguage = "\uD83C\uDDEC\uD83C\uDDE7"; //GB emoji
     }
 
+    @Override
+    public boolean equals(Object o) {
+        User user = (User) o;
+        return  !(o == null || getClass() != o.getClass()) &&
+                userID.equals(user.userID) &&
+                favoriteLocation.equals(user.favoriteLocation) &&
+                favoriteLanguage.equals(user.favoriteLanguage);
+    }
+
     public User(String name, String uId) {
         this.name = name;
         this.userID = new ID<>(uId);
         this.favoriteLocation = MapsHelper.ROLEX_LOCATION;
+
         this.favoriteLanguage = "\uD83C\uDDEC\uD83C\uDDE7"; //GB emoji
```

Comment:
I do not know if it already exists in the codebase but should not use hard-coded value. Perfectly fine in this PR tho

Reply:
I've previously added the class "Language" which contains all these hard-coded values. It's not a big deal to use it here